mesonbuild/compilers/vala.py

```diff
@@ -142,6 +142,46 @@ def thread_link_flags(self, env: 'Environment') -> T.List[str]:
     def get_option_link_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
         return []
 
+    def build_wrapper_args(self, env: 'Environment',
+                           extra_args: T.Union[None, CompilerArgs, T.List[str], T.Callable[[CompileCheckMode], T.List[str]]],
+                           dependencies: T.Optional[T.List['Dependency']],
+                           mode: CompileCheckMode = CompileCheckMode.COMPILE) -> CompilerArgs:
+        if callable(extra_args):
+            extra_args = extra_args(mode)
+        if extra_args is None:
+            extra_args = []
+        if dependencies is None:
+            dependencies = []
+
+        # Collect compiler arguments
+        args = self.compiler_args(self.get_compiler_check_args(mode))
+        for d in dependencies:
+            # Add compile flags needed by dependencies
+            if mode is CompileCheckMode.LINK and self.force_link:
+                # As we are passing the parameter to valac we don't need the dependent libraries.
+                a = d.get_compile_args()
+                if len(a) > 0:
```

Comment:
```suggestion
                if a:
```

Reply:
I didn't know you could do this. Thanks.